source/using-keycloak.rst

```diff
@@ -0,0 +1,392 @@
+several possible configurations.
+It is important to follow the currently established security standards for OAuth 2
+and OpenIDConnect.
+There are several keycloak libraries in various languages
+that can help with this development.
+
+For a frontend application
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We have a microservice in order to help the development to ensure more security
+for using keycloak with dojot, see more about `Backstage`_.
+You can use keycloak directly with `PKCE`_ and OpenID Connect as well.
+It is important to follow  the currently established security standards for
+OAuth 2 and OpenIDConnect.
+
+For a mobile application
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+We don't have use cases, but it's important to follow the currently
+established security standards for OAuth 2 and OpenIDConnect.
+Some starting points to pay attention to would be to create a
+new Client with the Access Type `public` and use `PKCE`_.
+
+Deployment Settings
+-------------------
+
+.. _Setting Passwords (Required) via Environment Variables:
+
+Setting Passwords (Required) via Environment Variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The dojot deployments (**Docker compose and Ansible - kubernetes**) do not come with
+passwords set for the ``admin`` and ``master`` users and it is necessary to set these
+passwords so that the *deployments* will start up correctly,
+if not configured services will not start and dojot will be unavailable.
+But beware these values are only applied when the keycloak starts
+for the first time and the *realm* are created, it will not affect the existing realms.
+
+.. attention::
+   When configuring use strong passwords, by default the minimum required are passwords
+   with 8 characters, one number, one uppercase letter,
+   one lowercase letter and one special character.
+
+
+Docker compose
+^^^^^^^^^^^^^^
+
+You need to set a password value in the *.env* file for the
+``KEYCLOAK_MASTER_PASSWORD`` and ``KEYCLOAK_ADMIN_PASSWORD_TEMP``
+variables. The ``KEYCLOAK_ADMIN_PASSWORD_TEMP`` value will be the *admin*
+user password of all realms when created.
+See more at `settings required`_.
+
+
+Ansible (kubernetes)
+^^^^^^^^^^^^^^^^^^^^
+
+*Work in progress*
+
+Setting SMTP via environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is possible to configure SMTP via environment variables,
+but beware these values are only applied when the *realm*
+is created, it will not affect the existing realms.
+To configure a *realm* that already exists,
+see `Configuring SMTP through the graphical interface`.
+
+
+Docker compose
+^^^^^^^^^^^^^^
+
+See more at `configuring SMTP`_.
+
+Ansible (kubernetes)
+^^^^^^^^^^^^^^^^^^^^
+
+*Work in progress*
+
+Setting HTTPs as required via environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Note: These values are only applied when the *realm* is created,
+it will not affect existing realms.
+For existing reals see `Configuring HTTPs as required`
```

Comment:
I don't know if it should be a link in `Configuring HTTPs as required`, but if it should, the link was not set.